### PR TITLE
Eli/xdr iocs fix

### PR DESCRIPTION
**Pull request by esharf**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
https://github.com/demisto/etc/issues/25403, nightly build.

## Minimum version of Demisto
- [ ] 4.5.0
- [ ] 5.0.0
- [x] 5.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [ ] Documentation

### CODE REVIEW

File: Packs/XDR_iocs/Integrations/XDR_iocs/XDR_iocs.yml [hunk 1]
```diff
@@ -122,9 +125,7 @@ script:
       required: false
       secret: false
     deprecated: false
-    description: |-
-      Run once when you configure the integration (do NOT run this twice!).
-      will all the indicators that were synced with XDR and then resync.
+    description: Sync your IOC with XDR and delete the old.
```
Comment (yaakovi) [inline Packs/XDR_iocs/Integrations/XDR_iocs/XDR_iocs.yml:128]:
```suggestion
    description: Sync your IOC with Cortex XDR and delete the old.
```

File: Packs/XDR_iocs/Integrations/XDR_iocs/XDR_iocs.py [hunk 2]
```diff
@@ -125,9 +125,10 @@ def create_file_sync(file_path, batch_size: int = 200):
         total_size: int = get_iocs_size()
         for i in range(0, ceil(total_size / batch_size)):
             iocs: List = get_iocs(page=i, size=batch_size)
-            for ioc in map(lambda x: json.dumps(demisto_ioc_to_xdr(x)), iocs):
-                _file.write(ioc)
-                _file.write('\n')
+            for ioc in map(lambda x: demisto_ioc_to_xdr(x), iocs):
+                if ioc:
+                    _file.write(json.dumps(ioc))
+                    _file.write('\n')
```
Comment (yaakovi) [inline Packs/XDR_iocs/Integrations/XDR_iocs/XDR_iocs.py:131]:
```suggestion
                    _file.write(json.dumps(ioc) + '\n')
```

File: Packs/XDR_iocs/Integrations/XDR_iocs/XDR_iocs.py [hunk 3]
```diff
@@ -246,19 +251,34 @@ def get_last_iocs(batch_size=200) -> List:
...
+        iocs = []
+        not_found = []
+        for indicator in indicators.split(','):
+            data = demisto.searchIndicators(value=indicator).get('iocs')
+            if data:
+                iocs.extend(data)
+            else:
+                not_found.append(indicator)
+        if not_found:
+            return_warning('the indicator{} {} was not found.'.format('s' if len(not_found) > 1 else '', ', '.join(not_found)))
```
Comment (yaakovi) [inline Packs/XDR_iocs/Integrations/XDR_iocs/XDR_iocs.py:265]:
```suggestion
            return_warning('The following indicators ware not found: '.format(', '.join(not_found)))
```

File: Packs/XDR_iocs/Integrations/XDR_iocs/XDR_iocs.py [hunk 4]
```diff
@@ -59,8 +60,11 @@ def http_request(self, url_suffix: str, requests_kwargs) -> Dict:
 
         if res.status_code in self.error_codes:
             raise DemistoException(f'{self.error_codes[res.status_code]}\t({res.content.decode()})')
-
-        return res.json()
+        try:
+            return res.json()
+        except json.decoder.JSONDecodeError as error:
+            demisto.error(str(res.content))
+            raise error
```
Comment (yaakovi) [inline Packs/XDR_iocs/Integrations/XDR_iocs/XDR_iocs.py:67]: why do you need this? unless you handle the error

File: Packs/XDR_iocs/Integrations/XDR_iocs/XDR_iocs.py [hunk 5]
```diff
@@ -179,26 +182,32 @@ def demisto_types_to_xdr(_type: str) -> str:
...
+        # get last 'IndicatorCommentRegular'
+        comment: Dict = next(filter(lambda x: x.get('type') == 'IndicatorCommentRegular', reversed(ioc.get('comments', []))), {})
+        if comment:
+            xdr_ioc['comment'] = comment.get('content')
+        if ioc.get('aggregatedReliability'):
+            xdr_ioc['reliability'] = ioc['aggregatedReliability'][0]
+        vendors = demisto_vendors_to_xdr(ioc.get('moduleToFeedMap', {}))
+        if vendors:
+            xdr_ioc['vendors'] = vendors
+        threat_type = ioc.get('CustomFields', {}).get('threattypes', {}).get('threatcategory', False)
```
Comment (yaakovi) [inline Packs/XDR_iocs/Integrations/XDR_iocs/XDR_iocs.py:202]: `threat type` is a boolean?

File: Packs/XDR_iocs/Integrations/XDR_iocs/XDR_iocs.py [hunk 6]
```diff
@@ -246,19 +255,34 @@ def get_last_iocs(batch_size=200) -> List:
...
+        iocs = []
+        not_found = []
+        for indicator in indicators.split(','):
+            data = demisto.searchIndicators(value=indicator).get('iocs')
+            if data:
+                iocs.extend(data)
+            else:
+                not_found.append(indicator)
+        if not_found:
+            return_warning('The following indicators ware not found: {}'.format(', '.join(not_found)))
```
Comment (yaakovi) [inline Packs/XDR_iocs/Integrations/XDR_iocs/XDR_iocs.py:269]: sorry, my typo
```suggestion
            return_warning('The following indicators were not found: {}'.format(', '.join(not_found)))
```

File: Packs/XDR_iocs/Integrations/XDR_iocs/XDR_iocs.py [hunk 7]
```diff
@@ -362,13 +386,34 @@ def create_iocs_to_keep_time():
     return hour, minute
 
 
+def is_xdr_data(ioc):
+    return ioc.get('sourceBrand') == 'Cortex XDR - IOC'
+
+
+def get_indicator_xdr_score(indicator: str, xdr_server: int):
```
Comment (yaakovi) [inline Packs/XDR_iocs/Integrations/XDR_iocs/XDR_iocs.py:393]: add docstring to this function

File: Packs/XDR_iocs/Integrations/XDR_iocs/XDR_iocs.py (hunk 4 above)
Comment (esharf) [reply Packs/XDR_iocs/Integrations/XDR_iocs/XDR_iocs.py:67]: not handling the error but this will keep us from wasting time on logs.

**Comment by yaakovi** [comment]
failed on on related error (MP problem)

**Pull request merged**